config/environments/development.rb

```diff
@@ -9,32 +9,54 @@
   # Do not eager load code on boot.
   config.eager_load = false
 
-  # Show full error reports and disable caching.
-  config.consider_all_requests_local       = true
-  config.action_controller.perform_caching = false
+  # Show full error reports.
+  config.consider_all_requests_local = true
+
+  # Enable/disable caching. By default caching is disabled.
+  # Run rails dev:cache to toggle caching.
+  if Rails.root.join('tmp', 'caching-dev.txt').exist?
+    config.action_controller.perform_caching = true
+
+    config.cache_store = :memory_store
+    config.public_file_server.headers = { 'Cache-Control' => "public, max-age=#{2.days.to_i}" }
+  else
+    config.action_controller.perform_caching = false
+
+    config.cache_store = :null_store
+  end
+
+  # Store uploaded files on the local file system (see config/storage.yml for options)
+  config.active_storage.service = :local
```

Comment:
Can we entirely remove ActiveStorage's config and not load it at all? I'd rather remove unused Rails components for speed + memory, and "saves files to the local filesystem" also makes me wary of security concerns. For example, disabling `DEFAULT_PARSERS` has saved us exposure to security bugs.

ActionCable, too, if we haven't pulled that out yet.

Reply:
Removed storage & cable. Unfortunately there is no clean way of doing this other than inlining the include file and deleting the individual lines.